libbeat/common/fleetmode/fleet_mode.go

```diff
@@ -32,18 +32,48 @@ func Enabled() bool {
 	}
 	var managementSettings management
 
-	cfgFlag := flag.Lookup("E")
-	if cfgFlag == nil {
+	found, cliCfg := fetchSettingsFlags()
+	if !found {
 		return false
 	}
 
-	cfgObject, _ := cfgFlag.Value.(*config.SettingsFlag)
-	cliCfg := cfgObject.Config()
-
 	err := cliCfg.Unpack(&managementSettings)
 	if err != nil {
 		return false
 	}
 
 	return managementSettings.Enabled
 }
+
+// ShipperMode checks to see if we're running filebeat as a shipper.
+// This is primarily used to disable global processors.
+// Handling global processors in the shipper is an ongoing issue, see https://github.com/elastic/elastic-agent-shipper/issues/292
+func ShipperMode() bool {
```

Comment:
Does this implementation mean that the shipper has to be enabled on the command-line rather than in the beat configuration? (This is ok if so, I'm just trying to understand what constraints we're working with.)

Reply:
So, technically it has to be both, since the global processor setup happens so early on that we have to catch it before we get any config from fleet.